parser_tool/static/js/src/spot_it.js

```diff
@@ -0,0 +1,482 @@
+(function ($) {
+  /**
+   * Theorem that given a prime number you could generate N * N + N + 1 cards with N + 1 symbols
+   * whose relationship between each card has a maximum of one related symbol
+   * Note: reason we use prime numbers is due to the theorem which uses a diagonalization method for generating the 1 to 1 relationship between the cards
+   * Ref: https://math.stackexchange.com/questions/1303497/what-is-the-algorithm-to-generate-the-cards-in-the-game-dobble-known-as-spo
+   **/
+  const generateDeckFromPrimeNum = (n) => {
```

Comment:
If I'm understanding correctly, there are 3 variables that are determined by the prime number passed to this function:

```
C = total number of cards in the deck
S = total number of symbols (or images)
N = number of symbols per card
```

So given a prime number `n`, we get `n+1` symbols per card, with `n^2 + n + 1` total cards in the deck and `n^2 + n + 1` total symbols. And the output of this function is an array of cards with the symbols on each, as represented by a number from `0...N`. Is that correct?

Reply:
Yes! Thats right! Something interesting is the pattern behind the algorithm. I still dont understand it 100%. I found a very cool visualization off of reddit below.
![spoti-it-reddit](https://user-images.githubusercontent.com/29421667/149998148-7e6ca175-228e-4d92-ac0b-f997f2f0c252.png)
.

Reply:
That's really neat! I'm going to have to study that a bit to make sense of it, but a very cool visualization.